src/pages/ca-checkbook/forms/CheckbookForm.js

```diff
@@ -0,0 +1,188 @@
+    ;(async function () {
+      try {
+        if (recordId) {
+          const res = await getRequest({
+            extension: CashBankRepository.CACheckbook.get,
+            parameters: `_recordId=${recordId}`
+          })
+
+          formik.setValues({
+            ...res.record,
+            issueDate: formatDateFromApi(res.record.issueDate)
+          })
+        }
+      } catch (exception) {}
+    })()
+  }, [])
+
+  return (
+    <FormShell resourceId={ResourceIds.Checkbook} form={formik} maxAccess={maxAccess} editMode={editMode}>
+      <VertLayout>
+        <Grow>
+          <Grid container spacing={4}>
+          <Grid item xs={12}>
+              <ResourceLookup
+                endpointId={CashBankRepository.CashAccount.snapshot}
+                parameters={{
+                  _type: 0
+                }}
+                required
+                name='bankAccountId'
+                label={labels.bank}
+                valueField='reference'
+                displayField='name'
+                valueShow='bankAccountRef'
+                secondValueShow='bankAccountName'
+                form={formik}
+                onChange={(event, newValue) => {
+                  formik.setFieldValue('bankAccountId', newValue?.recordId || '')
+                  formik.setFieldValue('bankAccountRef', newValue?.reference || '')
+                  formik.setFieldValue('bankAccountName', newValue?.name || '')
+                }}
+                error={formik.touched.bankAccountId && Boolean(formik.errors.bankAccountId)}
+                maxAccess={maxAccess}
+              />
+            </Grid>
+            <Grid item xs={12}>
+              <CustomTextField
+                name='firstCheckNo'
+                label={labels.firstCheckNo}
+                value={formik.values.firstCheckNo}
+                required
+                maxAccess={maxAccess}
+                onChange={formik.handleChange}
+                onClear={() => formik.setFieldValue('firstCheckNo', '')}
+                error={formik.touched.firstCheckNo && Boolean(formik.errors.firstCheckNo)}
+              />
+            </Grid>
+            <Grid item xs={12}>
+              <CustomTextField
+                name='lastCheckNo'
+                label={labels.lastCheckNo}
+                value={formik.values.lastCheckNo}
+                required
+                maxAccess={maxAccess}
+                onChange={formik.handleChange}
+                onClear={() => formik.setFieldValue('lastCheckNo', '')}
+                error={formik.touched.lastCheckNo && Boolean(formik.errors.lastCheckNo)}
+              />
+            </Grid>
+            <Grid item xs={12}>
+                <CustomNumberField
+                  name='size'
+                  type='text'
+                  required
+                  label={labels.size}
+                  value={formik.values.size}
+                  maxAccess={maxAccess}
+                  maxLength={100}
```

Comment:
maxLength used for checking number of digits entered & not for checking the value if exceeds 100